File: puzzles/python3/mime-type/mime_type.py
import sys

# dictionary to map file extensions to mime types
table = {}
nb_elements = int(input())
nb_names = int(input())
for i in range(nb_elements):
    extension, mime_type = input().split()
    table[extension.lower()] = mime_type

for i in range(nb_names):
    name = input().lower()
    dot_index = name.rfind(".")
    if dot_index == -1 or dot_index == len(name) - 1:
        print("UNKNOWN")
    else:
        extension = name[dot_index + 1:]
        try:
            print(table[extension])
        except KeyError:
            print("UNKNOWN")

File: puzzles/ts/mime-type/mime-type.ts
const nbElements: number = Number(readline()); // Number of elements which make up the association table.
const nbNames: number = Number(readline()); // Number of file names to be analyzed.
let table: object = {} // object to map file extensions to mime types
for (let i: number = 0; i < nbElements; i++) {
  const [extension, mimeType]: [string, string] = readline().split(' ');
  table[extension.toLowerCase()] = mimeType;
}

// For each of the filenames, display on a line the corresponding MIME type. If there is no corresponding type, then display UNKNOWN.
for (let i: number = 0; i < nbNames; i++) {
  const name: string = readline();
  const dotIndex: number = name.lastIndexOf('.');
  if (dotIndex === -1 || dotIndex === name.length - 1) {
    console.log("UNKNOWN");
  } else {
    const extension: string = name.substring(dotIndex + 1).toLowerCase();
    if (extension in table) {
      console.log(table[extension]);
    } else {
      console.log("UNKNOWN");
    }
  }
}
